src/preconnect.js

```diff
@@ -213,34 +233,63 @@ class PreconnectService {
     if (!this.features_.preload) {
       return;
     }
-    if (opt_preloadAs == 'document' && this.platform_.isSafari()) {
-      // Preloading documents currently does not work in Safari,
-      // because it
-      // - does not support preloading iframes
-      // - and uses a different cache for iframes (when loaded without
-      //   as attribute).
+    // For Safari, do not request a resource if it is not a type that we can
+    // set the "as" attribute for. Safari caches by url+type, so we need to use
+    // the correct type and cannot fall back to "fetch".
+    const isSafari = this.platform_.isIos() || this.platform_.isSafari();
+    if (isSafari && !this.isSupportedPreloadAs_(preloadAs)) {
       return;
```

Comment:
Maybe here we could still preconnect

Reply:
This should be handled 4 lines earlier via `this.url(viewer, url, /* opt_alsoConnecting */ true);` (though I would agree that function name can be confusing).